src/main/java/org/jboss/modules/ConcurrentClassLoader.java

```diff
@@ -455,7 +455,12 @@ protected Package definePackage(final String name, final String specTitle, final
             if (existing != null) {
                 return existing;
             }
-            Package pkg = super.definePackage(name, specTitle, specVersion, specVendor, implTitle, implVersion, implVendor, sealBase);
+            Package pkg;
+            try {
+                pkg = super.definePackage(name, specTitle, specVersion, specVendor, implTitle, implVersion, implVendor, sealBase);
+            } catch (final IllegalArgumentException iae) {
+                pkg = super.getPackage(name);
+            }
```

Comment:
I noticed that on JDK8 whole method ClassLoader.definePackage() was synchronized.
In JDK9 they're using ConcurrentHashMap.putIfAbsent() instead so it opens a window
for potential concurrent update of the underlying package map. I think this design change is the reason why we can see IllegalArgumentException in some tests.